vm_backtrace.c

```diff
@@ -517,16 +549,43 @@ backtrace_each(const rb_execution_context_t *ec,
 	  RUBY_VM_NEXT_CONTROL_FRAME(start_cfp)); /* skip top frames */
 
     if (start_cfp < last_cfp) {
-	size = 0;
+        size = last = 0;
     }
     else {
 	size = start_cfp - last_cfp + 1;
+
+        if (from_last > size) {
+            size = last = 0;
+            ret = 1;
```

Comment:
only 1 question. what does this condition mean?

Reply:
This is needed so that if you provide `caller` or `caller_locations` a starting frame larger than the total number of frames, you get nil instead of the empty array.

Reply:
I got it.